azure/scope/machine.go

```diff
@@ -221,6 +222,73 @@ func (m *MachineScope) InboundNatSpecs(portsInUse map[int32]struct{}) []azure.Re
 
 // NICSpecs returns the network interface specs.
 func (m *MachineScope) NICSpecs() []azure.ResourceSpecGetter {
+	nicSpecs := []azure.ResourceSpecGetter{}
+	defaultSpec := m.DefaultNICSpec()
+	if len(m.AzureMachine.Spec.NetworkInterfaces) < 1 {
+		return append(nicSpecs, defaultSpec)
+	}
+
+	for i, n := range m.AzureMachine.Spec.NetworkInterfaces {
+		var spec *networkinterfaces.NICSpec
+		if n.ID != "" {
+			continue
```

Comment:
this deviates significantly from the pattern we've established for other BYO existing resource in CAPZ. The way it works for other resources is that the reconciler will add a tag on each resource it creates to allow it to identify the resources it created vs. the ones the user pre-created. See the virtual networks as an example. There are pros and cons for both ways of doing it, however because this is creating an inconsistent user experience I would recommend putting BYO network interface as out of scope for this PR and tackling it in its own issue/PR so we can iterate on the design.